### PR TITLE
Initial Docker setup

### PR DESCRIPTION
**Pull request by bergie**
This should make it easier to develop the MsgFlo setup as you can run the whole thing locally with Docker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM node:6
+
+# Export the Websocket port for Flowhub connection
+EXPOSE 3569
+
+# Reduce npm install verbosity, overflows Travis CI log view
+ENV NPM_CONFIG_LOGLEVEL warn
+# Configure Bitraf MQTT
+ENV MSGFLO_BROKER mqtt://mqtt.bitraf.no
```
Comment (jonnor) [inline Dockerfile]: This should not be default, this is the production system, only one instance can use that.

Comment (bergie) [inline Dockerfile]: Fixed in 1642284

**Comment by bergie** [comment]
@jonnor added docs in fadd45c

**Comment by CISC** [comment]
Some suggestions:

1. Combine multiple ENV and RUN statements into one to minimize layers.
2. Since you use COPY on current directory, please add a .dockerignore file that excludes things like .git folder etc (or better yet, use a negated pattern to only include what is necessary)!

**Comment by bergie** [comment]
@CISC done

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+version: '2'
+services:
+  mqtt:
+    image: ansi/mosquitto
```
Comment (jonnor) [inline docker-compose.yml]: Why not using eclipse-mosquitto? Seemso be the offical iamge

Comment (bergie) [inline docker-compose.yml]: This is the one we've been using with the other MsgFlo docker setups, and we know it works. No other particular reason

**Comment by CISC** [comment]
@bergie That wasn't quite what I meant, you should have ENV as its own statement, but you have multiple RUN statements that you can merge into a single one.

**Comment by bergie** [comment]
Note: now Travis should build and run the Docker image. Also added fbp-spec for smoke testing.

**Pull request merged**